src/blob-api.js

```diff
@@ -0,0 +1,124 @@
+import fs from 'node:fs'
+import { basename } from 'node:path'
+import sodium from 'sodium-universal'
+import b4a from 'b4a'
+
+import { getPort } from './blob-server/index.js'
+
+/** @typedef {import('./types.js').BlobId} BlobId */
+/** @typedef {import('./types.js').BlobType} BlobType  */
+/** @typedef {import('./types.js').BlobVariant<BlobType>} BlobVariant  */
+
+export class BlobApi {
+  /**
+   * @param {object} options
+   * @param {string} options.projectId
+   * @param {import('./blob-store/index.js').BlobStore} options.blobStore
+   * @param {import('fastify').FastifyInstance} options.blobServer
+   */
+  constructor({ projectId, blobStore, blobServer }) {
+    this.projectId = projectId
+    this.blobStore = blobStore
+    this.blobServer = blobServer
+  }
+
+  /**
+   * Get a url for a blob based on its BlobId
+   * @param {import('./types.js').BlobId} blobId
+   * @returns {Promise<string>}
+   */
+  async getUrl(blobId) {
+    const { driveId, type, variant, name } = blobId
+    const port = await getPort(this.blobServer.server)
+    return `http://127.0.0.1:${port}/${this.projectId}/${driveId}/${type}/${variant}/${name}`
+  }
+
+  /**
+   * Write blobs for provided variants of a file
+   * @param {{ original: string, preview?: string, thumbnail?: string }} filepaths
+   * @param {{ mimeType: string }} metadata
+   * @returns {Promise<{ original: Omit<BlobId, 'driveId'>, preview?: Omit<BlobId, 'driveId'>, thumbnail?: Omit<BlobId, 'driveId'> }>}
```

Comment:
Ah very good point. It's actually a property on the writeStream e.g. `blobStore.createWriteStream().driveId` although I just realised that was not documented (my bad) and it's an awkward API too. Since the driveId for any written blobs will never change for a particular instance of the blobStore, I think it makes sense to just expose it as `blobStore.writerDriveId`, and use it from there. I've created [an issue](https://github.com/digidem/mapeo-core-next/issues/218) and [made a fix](https://github.com/digidem/mapeo-core-next/pull/219)